src/qibo/backends/numpy.py

```diff
@@ -328,6 +333,36 @@ def thermal_error_density_matrix(self, gate, state, nqubits):
         state = self.apply_gate(gate, state.ravel(), 2 * nqubits)
         return self.np.reshape(state, shape)
 
+    def depolarizing_error_density_matrix(self, gate, state, nqubits):
```

Comment:
Maybe add one-line comments to explain what some of the variables are (e.g. `qubits_1`, `qubits_2`, `qs`) and why they are needed.

Reply:
Those variables are used to write the tensor product $Tr_q(\rho)\otimes I^{\otimes n}$ in the correct order. First, I calculate the tensor product using `np.tensordot()` as if the depolarization channel acted on the last `len(q)` qubits, and then I change the order.